scripts/firedrake-install

```diff
@@ -2004,6 +2014,14 @@ else:
             f90 = os.path.join(compilerbin, "mpif90")
             compiler_env = create_compiler_env(cc, cxx, f90)
 
+    # Make sure that we link against the right MPI and PETSc shared libraries
+    link_env = {
+        "LDFLAGS": (
+            "-Wl,-rpath,{petsc_dir}/{petsc_arch}/lib -L{petsc_dir}/{petsc_arch}/lib"
+            .format(petsc_dir=petsc_dir, petsc_arch=petsc_arch)
+        ),
+    }
```

Comment:
Is there a way to avoid this block given it's identical to L1882-L1889?

Reply:
Maybe. But I honestly think it might be clearer to put here next to `compiler_env` since they are very similar and that code is already duplicated.